### DISCUSSION TITLE
What is the actual difference in implementation between @layer base, components and utilities directives?

### DISCUSSION THREAD

**Discussion by nemanjam** [tailwindlabs/tailwindcss#13023] [Help]
Here from docs I can understand that `@layer base` is meant for tags i.e. `h1, a, p...`, `@layer components` is meant for component level classes and `@layer utilities` has the highest specificity. But then I have also seen classes defined in `@layer base` and this confuses me. Also there are regular CSS rules that are defined outside of any layer.

And beside the specificity I don't understand what is the actual difference between these layers, will the rules from each layer be processed in different way and end up in different place in CSS output? What is the actual difference in output for each layer?

https://tailwindcss.com/docs/functions-and-directives#layer

Can someone clarify this a bit? Docs explains the purpose but not so much the actual processing mechanism.

**Comment by wongjn** [comment]
> And beside the specificity […]

Writing a CSS rule in a `@layer` does not affect its specificity in and of itself.

> […] what is the actual difference between these layers […]

They map to the different `@tailwind base/components/utility` statements.

> […] will the rules from each layer be processed in different way […]

I don't believe so.

> […] will the rules from each layer […] end up in different place in CSS output?

Yes, they will be placed in the relevant `@tailwind base/components/utility` statement location.

> What is the actual difference in output for each layer?

See above.
